sdk/python/kfp/utils.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def use_secret(secret_name:str, secret_volume_mount_path:str, volume_name:str=None, env_variable:str=None, secret_file_path_in_volume:str=None):
```

Comment:
Is there any need for the `volume_name`? It's probably best if it's generated so that it does not clash with other volumes.

Reply:
When looking at the implementations for (gcp)[https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/gcp.py] the volume is also stated. I though it was best to allow the user to set it. Also to maybe later align to use a common underlying implementation that the different cloud vendor specific implementations could use to have a common code base.

Reply:
I saw that this will actually change for gcp as well. I will updated to auto generate it :)